Sources/ArcGISToolkit/Components/UtilityNetworkTrace/UtilityNetworkTraceViewModel.swift

```diff
@@ -115,27 +115,22 @@ import SwiftUI
         }
     }
     
-    /// Adds a new starting point to the pending trace.
+    /// Adds new starting points to the pending trace.
     /// - Parameters:
     ///   - point: A point on the map in screen coordinates.
     ///   - mapPoint: A point on the map in map coordinates.
     ///   - proxy: Provides a method of layer identification.
-    func addStartingPoint(
-        at point: CGPoint,
-        mapPoint: Point,
-        with proxy: MapViewProxy
-    ) async {
-        let identifyLayerResults = try? await proxy.identifyLayers(
-            screenPoint: point,
-            tolerance: 10
-        )
-        for layerResult in identifyLayerResults ?? [] {
-            for geoElement in layerResult.geoElements {
-                let startingPoint = UtilityNetworkTraceStartingPoint(
-                    geoElement: geoElement,
-                    mapPoint: mapPoint
-                )
-                await processAndAdd(startingPoint: startingPoint)
+    ///
+    /// An identify operation will run on each layer in the network. Every element returned from
+    /// each layer will be added as a new starting point.
+    func addStartingPoints(at point: CGPoint, mapPoint: Point, with proxy: MapViewProxy) async {
+        for layer in network?.layers ?? [] {
+            if let result = try? await proxy.identify(on: layer, screenPoint: point, tolerance: 10) {
+                for element in result.geoElements {
+                    await processAndAdd(
+                        startingPoint: UtilityNetworkTraceStartingPoint(geoElement: element, mapPoint: mapPoint)
+                    )
+                }
             }
         }
```

Comment:
Currently, calling `identify` on each layer will wait until `identify` is done before moving on to the next layer. You can use a `TaskGroup` to perform all identifies at the same time.  I've used it once before; I'm not sure if the `try? await group.waitForAll()` is necessary (or i you need another one in the inner task group); also, I did not make sure this compiles.

I thought about another task group around the `await processAndAdd` for loop, but there's a lot of stuff happening inside there, including setting the error alert, and I'm not sure what the consequences of performing those all asynchronously would be, but it may be worth a shot.

```suggestion
        await withTaskGroup(of: Void.self) { [weak self] taskGroup in
            guard let self else { return }
            for layer in network?.layers ?? [] {
                taskGroup.addTask {
                    if let result = try? await proxy.identify(on: layer, screenPoint: point, tolerance: 10) {
                        for element in result.geoElements {
                            await processAndAdd(
                                startingPoint: UtilityNetworkTraceStartingPoint(geoElement: element, mapPoint: mapPoint)
                            )
                        }
                    }
                }
            }
            // Make sure that all tasks complete.
            try await group.waitForAll()
        }
```

Reply:
Good suggestion, applied, thank you!